WalletWasabi.Fluent/ViewModels/Wallets/Send/PrivacySuggestionsFlyoutViewModel.cs

```diff
@@ -28,6 +34,27 @@ public PrivacySuggestionsFlyoutViewModel(Wallet wallet)
 
 	public ObservableCollection<PrivacyWarning> Warnings { get; } = new();
 	public ObservableCollection<PrivacySuggestion> Suggestions { get; } = new();
+	public IObservable<IEnumerable<PrivacyWarning>> PreviewWarnings => _previewWarnings;
+
+	public async Task UpdatePreviewWarningsAsync(TransactionInfo info, BuildTransactionResult transaction, CancellationToken cancellationToken)
+	{
+		var previewWarningList = new List<PrivacyWarning>();
+
+		await foreach (var item in _privacySuggestionsModel.BuildPrivacySuggestionsAsync(info, transaction, cancellationToken))
+		{
+			if (item is PrivacyWarning warning)
+			{
+				previewWarningList.Add(warning);
+			}
+		}
```

Comment:
The calculation of Warnings and Suggestions at the same time feels unnecessary here. Only the warnings are important.

Can we separate the warning generation from the suggestion generation?
So in `BuildPrivacySuggestionsAsync()` we could do:
```
var warnings = GetWarnings();
var suggestions = GetSuggestions(warnings);
```

And in `UpdatePreviewWarningsAsync()` we could simply use `GetWarnings()`.

What do you think?

Reply:
@soosr separating Warnings from Suggestions would require significant refactoring (and probably lots of duplicated code) in methods such as `PrivacySuggestionsModel.VerifyPrivacyLevel()`. Just by looking at the code it is evident that it uses a lot of logic and variables in calculating both Warnings *and* Suggestions.

What I can do is to use a `bool includeSuggestions = true` parameter, and if that is false we return early from the iterator methods. Does that make sense?

Reply:
Yes, that would be a huge improvement. Suggestions are basically transactions that needs to be built. But that is time consuming so we should skip it if possible.

Reply:
@soosr pushed ;)